### PR TITLE
Update 2024

### PR DESCRIPTION
**Pull request by esarrazin**
@guillaumeeb, I've updated the Python part of the course. Let me know what you think. I still need to make/update the tutorials. 
Do you want to keep the quizzes? If so, I'll have to update the links. For the new courses, I haven't put any questions.

### CODE REVIEW

File: src/24_Large_Datasets.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  - Version 2: 1,692,919,135
+  - Version 3: 1,811,709,771
+
+:::
+::: {.column width="50%"}
+
+### EUCLID
+
+- Mission: Explore the composition and evolution of the dark Universe
+- Launch in 2023
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md]:
```suggestion
- Launched in 2023
```

File: src/24_Large_Datasets.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+- Data volume: 170 PB (6 years) i.e. 80 TB / day
+
+:::
+::::::::::::::
+
+## Larger datasets
+
+- Training large deep neural networks required large datasets
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md]: requires?

File: src/24_Large_Datasets.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+## Zarr
+
+![](images/zarr_logo.png)
+
+- Free and open-source
+- Array oriented file format
+- File storage format for chunked, compressed, N-dimensional arrays
+- Same performances than HDF5
+- More flexible because chunking can be done along any dimension
+
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md:113]: Cloud optimized?

File: src/24_Large_Datasets.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+- the type of access: partial access or global access
+- the usage: compatibility with the library or the application used
+
+
+# Where to store large datasets ?
+
+## Distributed data storage
+
+- HDFS
+- Amazon S3
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md:149]: Object storage: ...

File: src/24_Large_Datasets.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+## Good pratices: Use efficient datatypes
+
+:::::::::::::: {.columns}
+::: {.column width="50%"}
+
+- Objective: Reduce memory foot-print
+- Text data column is not memory efficient
+- Use pandas.Categorical for categorical data
+- Use pandas.numeric() to downcast the numeric columns to their smallest types 
+
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md:182]: PyArrow Strings in Pandas 2?

File: src/24_Large_Datasets.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+:::
+::::::::::::::
+
+## Good pratices: Dask
+
+- Dask includes dask.dataframe, a pandas-like API for working with larger than memory datasets in parallel. 
+- Dask can use multiple threads or processes on a single machine, or a cluster of machines to process data in parallel.
+- More information in the next course
+
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md:224]: Use it only if needed, Pandas is often the best for dataset that fits in memory

File: src/23_Large_Datasets.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
+---
```
Comment (guillaumeeb) [inline src/23_Large_Datasets.md]: I think this file can be deleted

**Review by guillaumeeb — COMMENTED** [review]
Thanks a lot! A few comments on the content. Do as you want foor the quizz!

File: src/23_Large_Datasets.md (hunk 7 above)
Comment (esarrazin) [reply src/23_Large_Datasets.md]: Thanks, I change the order and clearly I renamed the file incorrectly

File: src/24_Large_Datasets.md (hunk 2 above)
Comment (esarrazin) [reply src/24_Large_Datasets.md]: Ok I remove this slide

File: src/24_Large_Datasets.md (hunk 5 above)
Comment (esarrazin) [reply src/24_Large_Datasets.md:182]: Ok I didn't know that.

File: src/24_Large_Datasets.md (hunk 6 above)
Comment (esarrazin) [reply src/24_Large_Datasets.md:224]: Ok

**Comment by esarrazin** [comment]
Thanks for the proofreading ! If the rest is ok for you, I'll merge. I will update the links for the quizzes during the week

File: src/24_Large_Datasets.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+- the type of access: partial access or global access
+- the usage: compatibility with the library or the application used
+
+
+# Where to store large datasets ?
+
+## Distributed data storage
+
+Object storage:
+- HDFS
```
Comment (guillaumeeb) [inline src/24_Large_Datasets.md:148]: I would not consider HDFS as an object storage!

**Review by guillaumeeb — COMMENTED** [review]
Just HDFS outside of Object storage

**Comment by guillaumeeb** [comment]
Okay, I'll merge as is, and maybe do the correction on HDFS later on.

**Pull request merged**